src/services/review/RepoCheckerService.ts

```diff
@@ -4,110 +4,358 @@ import path from "path"
+      .orElse((errors) => {
+        logger.error(`Error running site checker for repos: ${errors}`)
+        return errAsync(errors)
+      })
+  }
+
+  /**
+   * Run the site checker for all repos sequentially
+   * Not using neverthrow here as we want to run all the checkers
+   * in sequence without short circuiting, and neverthrow does not
+   * support this.
+   */
+  async runCheckerForReposInSeq(repos: string[]): Promise<void> {
+    for (const repoName of repos) {
+      const result = await this.checkRepo(repoName)
+      if (result.isErr()) {
+        logger.error(`Something went wrong when checking repo: ${result.error}`)
+      }
+    }
+  }
+
+  checkRepo(repo: string): ResultAsync<PushResult, SiteCheckerError> {
     const repoPath = path.join(EFS_VOL_PATH_STAGING, repo)
+    /**
+     * To allow for an easy running of the script, we can temporarily clone
+     * the repo to the EFS
+     */
+    let repoExists = true
 
     const mdFiles = new Set<string>()
-    // get list of all documents
-    const checker = this.getListOfMarkdownFiles(repoPath)
-      .andThen((files) => {
-        files.forEach((file) => {
-          mdFiles.add(file)
+
+    const errors: RepoError[] = []
+    let stagingLink: string
+    const getStagingLink = this.getStagingLinkFromRepoName(repo).andThen(
+      (stagingLinkFromRepo) => {
+        stagingLink = stagingLinkFromRepo
+        return ok(stagingLinkFromRepo)
+      }
+    )
+
+    const cloner = () =>
+      this.gitFileSystemService
+        .isValidGitRepo(repo, STAGING_BRANCH)
+        .andThen((isValidGitRepo) => {
+          // todo: check if this check works properly in staging and production env, seems to
+          // hit even when local has the necessary repo, not too sure why
+          logger.info(
+            `Repo ${repo} does not exist in efs for site checker, cloning...`
+          )
+          // it is ok if file does not exist, just clone it for now
+          repoExists = isValidGitRepo
+          return this.gitFileSystemService
+            .cloneBranch(repo, true)
+            .map(() => true)
+            .mapErr((error) => new SiteCheckerError(`${error}`))
         })
-        console.log("calling getAllDocumentsPath", repoPath)
-        return this.getAllMediaPath(repoPath)
-      })
-      .andThen((setOfAllMediaPath) => {
-        console.log(setOfAllMediaPath)
-
-        return ResultAsync.fromPromise(
-          this.getAllPermalinks(repoPath, mdFiles),
-          (error) => new SiteCheckerError(`${error}`)
-        ).andThen((setOfAllPermalinks) =>
-          ok(new Set([...setOfAllMediaPath, ...setOfAllPermalinks]))
+        .mapErr((error) => new SiteCheckerError(`${error}`))
+
+    const remover = () => {
+      if (!repoExists) {
+        return this.gitFileSystemService
+          .removeRepo(repo, STAGING_BRANCH)
+          .mapErr((error) => new SiteCheckerError(`${error}`))
+      }
+      return okAsync(undefined)
+    }
+
+    const checker = () =>
```

Comment:
i think for each submethod here, we probably want to separate them out into their own method so the dependencies are clear.